### PR TITLE
CTCP-3335: create add transport charges yes no page

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/pages/item/AddTransportChargesYesNoPage.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import pages.QuestionPage
+import pages.sections.ItemSection
+import play.api.libs.json.JsPath
+import play.api.mvc.Call
+
+case class AddTransportChargesYesNoPage(itemIndex: Index) extends QuestionPage[Boolean] {
+
+  override def path: JsPath = ItemSection(itemIndex).path \ toString
+
+  override def toString: String = "transportCharges"
```
Comment (jackheslop96) [inline app/pages/item/AddTransportChargesYesNoPage.scala:30]: Might be worth making this "addTransportChargesYesNo" just incase it conflicts with the path of any subsequent pages that get built

File: conf/app.item.routes [hunk 2]
```diff
@@ -125,3 +125,8 @@ GET        /:itemIndex/transport-equipment/:lrn                             cont
...
 POST       /:itemIndex/change-transport-equipment/:lrn                      controllers.item.TransportEquipmentController.onSubmit(lrn: LocalReferenceNumber, mode: Mode = CheckMode, itemIndex: Index)
+
+GET        /:itemIndex/payment-method/:lrn                            controllers.item.AddTransportChargesYesNoController.onPageLoad(lrn: LocalReferenceNumber, mode: Mode = NormalMode, itemIndex: Index)
+POST       /:itemIndex/payment-method/:lrn                            controllers.item.AddTransportChargesYesNoController.onSubmit(lrn: LocalReferenceNumber, mode: Mode = NormalMode, itemIndex: Index)
+GET        /:itemIndex/change-payment-method/:lrn                     controllers.item.AddTransportChargesYesNoController.onPageLoad(lrn: LocalReferenceNumber, mode: Mode = CheckMode, itemIndex: Index)
+POST       /:itemIndex/change-payment-method/:lrn                     controllers.item.AddTransportChargesYesNoController.onSubmit(lrn: LocalReferenceNumber, mode: Mode = CheckMode, itemIndex: Index)
```
Comment (jackheslop96) [inline conf/app.item.routes:132]: Content doc indicates these should be `:itemIndex/payment-method/add/:lrn` (and `:itemIndex/change-payment-method/add/:lrn`)

File: conf/messages.en [hunk 3]
```diff
@@ -463,3 +463,10 @@ item.transportEquipment.placeholder = Select a transport equipment
 item.transportEquipment.change.hidden = transport equipment for item {0}
 item.transportEquipment.checkYourAnswersLabel = Transport equipment
 item.transportEquipment.error.required = Select which transport equipment you are using for this item
+
+item.addTransportChargesYesNo.title = Do you want to add a method of payment for this item's transport charges?
+item.addTransportChargesYesNo.heading = Do you want to add a method of payment for this item's transport charges?
+item.addTransportChargesYesNo.paragraph = These include charges for hauliers and carriers.
+item.addTransportChargesYesNo.checkYourAnswersLabel = Do you want to add a method of payment for this item's transport charges?
+item.addTransportChargesYesNo.hidden = if you want to add a method of payment for this item's transport charges
+item.addTransportChargesYesNo.error.required = Select yes if you want to add a method of payment for this item's transport charges
```
Comment (jackheslop96) [inline conf/messages.en:472]: Incorrect apostrophes being used. Should be `’` instead of `'`'

File: test/views/item/AddAnotherItemViewSpec.scala [hunk 4]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package views
+package views.item
```
Comment (jackheslop96) [inline test/views/item/AddAnotherItemViewSpec.scala:17]: Accidental diff? Doesn't seem to correlate with the location of the view its testing

File: app/pages/sections/PaymentMethodSection.scala [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.sections
+
+import play.api.libs.json.JsPath
+import pages.QuestionPage
+
+case object PaymentMethodSection extends QuestionPage[Nothing] {
```
[the remark this answers is not in the document]
Comment (kenneth-rayner) [reply app/pages/sections/PaymentMethodSection.scala:22]: Gone

File: app/pages/item/AddTransportChargesYesNoPage.scala (hunk 1 above)
Comment (kenneth-rayner) [reply app/pages/item/AddTransportChargesYesNoPage.scala:30]: Done

File: conf/messages.en (hunk 3 above)
Comment (kenneth-rayner) [reply conf/messages.en:472]: Done

**Comment by kenneth-rayner** [comment]
Test this please

File: conf/messages.en [hunk 6]
```diff
@@ -463,3 +463,10 @@ item.transportEquipment.placeholder = Select a transport equipment
 item.transportEquipment.change.hidden = transport equipment for item {0}
 item.transportEquipment.checkYourAnswersLabel = Transport equipment
 item.transportEquipment.error.required = Select which transport equipment you are using for this item
+
+item.addTransportChargesYesNo.title = Do you want to add a method of payment for this item’s transport charges?
+item.addTransportChargesYesNo.heading = Do you want to add a method of payment for this tem’s transport charges?
```
Comment (jackheslop96) [inline conf/messages.en:468]: Missing `i`

File: conf/messages.en [hunk 7]
```diff
@@ -463,3 +463,10 @@ item.transportEquipment.placeholder = Select a transport equipment
...
+item.addTransportChargesYesNo.paragraph = These include charges for hauliers and carriers.
+item.addTransportChargesYesNo.checkYourAnswersLabel = Do you want to add a method of payment for this item’s transport charges?
+item.addTransportChargesYesNo.hidden = if you want to add a method of payment for this tem’s transport charges
```
Comment (jackheslop96) [inline conf/messages.en:471]: Missing `i`

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**